### PR TITLE
Fix TypeScript top level await

### PR DESCRIPTION
**Pull request by AlCalzone**
```
// global script:
await wait(2000);
log("3");

// test script:
log("1");
await wait(1000);
log("2");
```
console output:
grafik

fixes: #669

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -791,7 +790,7 @@ function main() {
                                     adapter.log.info(obj._id + ': compiling TypeScript source...');
                                     // Force TypeScript to treat the code as a module.
                                     // Without this, it may complain about different scripts using the same variables.
-                                    const sourceWithExport = obj.common.source + '\nexport {};';
+                                    const sourceWithExport = `(async () => {${obj.common.source}})();\nexport {};`;
```
Comment (AlCalzone) [inline main.js:793]: Wrap the compilation input in an async function, not the compiled code below.

File: main.js [hunk 2]
```diff
@@ -813,7 +812,7 @@ function main() {
                                     } else {
                                         // We don't have a hashed source code or the original source changed, compile it
                                         const filename = scriptIdToTSFilename(obj._id);
-                                        const tsCompiled = tsServer.compile(filename, obj.common.source);
+                                        const tsCompiled = tsServer.compile(filename, sourceWithExport);
```
Comment (AlCalzone) [inline main.js:815]: don't forget to compile the wrapped code

File: main.js [hunk 3]
```diff
@@ -1587,9 +1586,9 @@ function prepareScript(obj, callback) {
         } else if (obj.common.engineType.toLowerCase().startsWith('typescript')) {
             // TypeScript
             adapter.log.info(name + ': compiling TypeScript source...');
-            // Force TypeScript to treat the code as a module.
+            // Force TypeScript to treat the code as a module by adding an empty export.
             // Without this, it may complain about different scripts using the same variables.
-            const sourceWithExport = `(async () => {${obj.common.source}\n})();`;
+            const sourceWithExport = `(async () => {${obj.common.source}})();\nexport {};`;
```
Comment (AlCalzone) [inline main.js:1591]: Wrap the compilation input in an async function, not the compiled code.

**Comment by GermanBluefox** [comment]
Thank you. Will try

**Pull request merged**